src/subdomains/core/aml/services/aml-helper.service.ts

```diff
@@ -230,6 +232,23 @@ export class AmlHelperService {
     return undefined;
   }
 
+  static amlRuleQuoteCheck(amlRules: AmlRule[], user: User, paymentMethodIn: PaymentMethod): QuoteError | undefined {
+    if (
+      user?.status === UserStatus.NA &&
+      ((amlRules.includes(AmlRule.RULE_2) && user?.userData.kycLevel < KycLevel.LEVEL_30) ||
+        (amlRules.includes(AmlRule.RULE_3) && user?.userData.kycLevel < KycLevel.LEVEL_50) ||
+        (amlRules.includes(AmlRule.RULE_6) &&
+          paymentMethodIn === FiatPaymentMethod.CARD &&
+          user?.userData.kycLevel < KycLevel.LEVEL_30) ||
+        (amlRules.includes(AmlRule.RULE_7) &&
+          paymentMethodIn === FiatPaymentMethod.CARD &&
+          user?.userData.kycLevel < KycLevel.LEVEL_50))
+    )
+      return QuoteError.KYC_REQUIRED;
+
+    return undefined;
+  }
```

Comment:
I would just return a boolean here, to keep the `QuoteError`s in the transaction helper. And adapt the method name accordingly 😉